src/staking/manager.ts

```diff
@@ -0,0 +1,660 @@
+   * @param stakingParamVersion - The param version that was used to create the EOI
+   * @param earlyUnbondingTx - The early unbonding transaction.
+   * @param feeRate - The fee rate in satoshis per byte.
+   * @returns The signed withdrawal transaction.
+   */
+  async createWithdrawEarlyUnbondedBtcTransaction(
+    stakingInput: StakingInputs,
+    stakingParamVersion: number,
+    earlyUnbondingTx: Transaction,
+    feeRate: number,
+  ): Promise<Transaction> {
+    const p = getBbnParamByVersion(stakingParamVersion, this.stakingParams);
+    if (!p) {
+      throw new Error(
+        `Unable to find staking params for version ${stakingParamVersion}`,
+      );
+    }
+
+    const stakerInfo = await this.btcProvider.getStakerInfo();
+    if (!stakerInfo) {
+      throw new Error("Staker info not found while creating signed unbonding transaction");
+    }
+
+    const staking = new Staking(
+      this.network,
+      stakerInfo,
+      p,
+      stakingInput.finalityProviderPkNoCoordHex,
+      stakingInput.stakingTimelock,
+    );
+
+    
+    const { psbt: unbondingPsbt } =
+    staking.createWithdrawEarlyUnbondedTransaction(
+      earlyUnbondingTx,
+      feeRate,
+    );
+
+    const signedWithdrawalPsbtHex = await this.btcProvider.signPsbt(
+      unbondingPsbt.toHex()
+    );
+    return Psbt.fromHex(signedWithdrawalPsbtHex).extractTransaction();
+  }
+
+  /**
+   * Creates a signed withdrawal transaction on the staking expired path that is 
+   * ready to be sent to the BTC network.
+   * @param stakingInput - The staking inputs.
+   * @param stakingParamVersion - The param version that was used to create the EOI
+   * @param stakingTx - The staking transaction.
+   * @param feeRate - The fee rate in satoshis per byte.
+   * @returns The signed withdrawal transaction.
+   */
+  async createWithdrawStakingExpiredBtcTransaction(
+    stakingInput: StakingInputs,
+    stakingParamVersion: number,
+    stakingTx: Transaction,
+    feeRate: number,
+  ): Promise<Transaction> {
+    const p = getBbnParamByVersion(stakingParamVersion, this.stakingParams);
+    if (!p) {
+      throw new Error(`Unable to find staking params for version ${stakingParamVersion}`);
+    }
+
+    const stakerInfo = await this.btcProvider.getStakerInfo();
+    if (!stakerInfo) {
+      throw new Error("Staker info not found while creating signed unbonding transaction");
+    }
+
+    const staking = new Staking(
+      this.network,
+      stakerInfo,
+      p,
+      stakingInput.finalityProviderPkNoCoordHex,
+      stakingInput.stakingTimelock,
+    );
+
+    const { psbt } = staking.createWithdrawStakingExpiredTransaction(
+      stakingTx,
+      feeRate,
+    );
+
+    const signedWithdrawalPsbtHex = await this.btcProvider.signPsbt(
+      psbt.toHex()
+    );
+    return Psbt.fromHex(signedWithdrawalPsbtHex).extractTransaction();
+  }
+
+  /**
+   * Creates a signed withdrawal transaction on the slashing path that is 
```

Comment:
Hm this is more an withdrawal from and expired change output of the slashing transaction